ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -2771,5 +2771,50 @@
     "name": "type-blank.svg",
     "mediaType": "image/svg+xml"
   }
- }
+  },
+  {
+    "id": "openshift",
+    "name": "openshift",
+    "description": "Openshift Workspace",
+    "scope": "general",
+    "tags": [
+      "Openshift"
+    ],
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "machines": {
+            "dev-machine/main": {
+              "installers": [
+                "org.eclipse.che.exec", "org.eclipse.che.terminal", "org.eclipse.che.ws-agent"
+              ],
+              "servers": {},
+              "attributes" : {
+                "memoryLimitBytes": "2147483648"
+              }
+            }
+          },
+          "recipe": {
```

Comment:
I don't think it's a bad idea to introduce a new recipe type `openshift`. 

But yes the approach we had with the current OpenShift connector was to be able to reuse current "docker image" recipes. It makes sense to run a `centos_jdk8` Docker image on OpenShift so we didn't feel the need to support the openshift deployment descriptions (yaml or json). 

The only drawback with your approach is that you will need to duplicate all the stacks in `stacks.json` if you want them to run on openshift.